05_iigo/index.tex

```diff
@@ -429,51 +452,522 @@ \section{IIGO Session order}
+Such division gives agents the freedom to implement the presidential functions as they like, provided that they follow the specified interface. Some agents may choose to obey the rules to ensure that the Judge inspecting their actions will never find any disobedience occurred. When implementing their presidential functions, islands can either follow a rule-obeying approach or try to exploit their presidential power when they are elected to be the President. For example, the President gets access to the reported amounts of private pool resources from all the islands and therefore, can use this confidential information for its own benefit in foraging or gifting.
 
-\section{Future Work}
+\subsection{Server-side Implementation}
+\label{sub:president:server-side}
+Server implementation of executive branch provides the President with the framework to use its powers of carrying out the law at the archipelago. The server is responsible for the vast majority of data processing and acts as a bridge between the President and the other agents. For example, the President decides on the tax distribution for the given turn, and the server converts this information into communication messages and broadcasts them to the islands. This ensures that the backend 'logistics' of the game, such as the format of messages passed between islands, remain the same between different President implementation. It also reduces the complexity of agents and minimises the amount of work for each of the team implementing their agents.
+
+The flow of every server-side function call of the executive branch can be presented in the following way:
+\begin{enumerate}
+    \item The required information from agents or other roles is obtained. This information includes \emph{allocation requests}, \emph{private pool resource reports} and \emph{rules} proposed by the islands.
+    \item This information is aggregated into a format accepted by the interface of presidential functions (usually in form $map[agent] \xrightarrow[]{} information$).
+    \item Presidential function is called only on the currently residing President with aggregated information as inputs.
+    \item The outcome of the presidential function call is processed.
+    \item The outcome of the action, or lack thereof, is passed to the members of the archipelago.
+\end{enumerate}
 
+Moreover, the server acts as a guard of the game state and prevents the President from performing game-breaking actions. For example, server-side of executive branch ensures that the President is only allowed to consider taking any action only if the monetary constraints allow. In other words, if the amount of resources in the common pool is lower than the action cost, then the President is not allowed to perform this action, as it would result in a negative amount of resources in the common pool. However, if the common pool holds enough resources, then it is up to the island implementation to decide whether the President will act on its powers.
```

Comment:
Repeated info, might just want to reference to the section above where this is explained already

Reply:
lollllllllllllg good catch!!!

we forgot to delete it or smth